### DISCUSSION TITLE
How to perform nested field validation check on create? Suggest some alternative.

### DISCUSSION THREAD

**Discussion by sarthakgaur** [directus/directus#7499] [Archive]
Let's say I have a role called "User Admin".

I want to give permission to "User Admin" to create users. But, the users "User Admin" creates should be restricted to the set of roles that I specify.

Right now, in `Settings > Role & Permissions > Directus Users + CREATE > Use Custom > Field Validation` I have this:

```
{
    "role": {
        "name": {
            "_in": [
                "Developer",
                "Manager",
                "Accountant"
            ]
        }
    }
}
```
So, the users created by "User Admin" can only have the roles "Developer", "Manager", and "Accountant". This validation gives me an `Invalid undefined schema`. I learned from this discussion(#6616) that the validation step currently doesn't handle nested fields.

If I can't do this, can someone suggest me an alternative way to do this? Any help will be appreciated! Thanks!

**Comment by rijkvanzanten** [comment]
> If I can't do this, can someone suggest me an alternative way to do this? Any help will be appreciated!

Heya! You can simply use the roles primary keys instead of nested name field 🙂 

```json
{
    "role": {
        "_in": [
            "d028c314-e9ee-4357-b7b8-8de47b7b8808",
            "943ae23d-8a60-49c2-a4c4-6d5ffe0b65c7",
            "ae155544-04a9-403b-9de4-fbe4fd198406"
        ]
    }
}
```
